ci/travis_install_osx.sh

```diff
@@ -30,6 +30,11 @@ if [ "$ARROW_CI_RUBY_AFFECTED" = "1" ]; then
             false
         fi
     }
+    # ARROW-3976 Old versions of git can cause failures when Homebrew prints a
+    # donation solicitation. Attempt to update git
+    git --version
+    run_brew install git
```

Comment:
`brew install` doesn't work for installed package.
We need to use `run_brew upgrade git`.